src/Interactive/Host/Interactive/Core/InteractiveHost.cs

```
@@ -413,25 +369,38 @@ private async Task<InitializedRemoteService> TryGetOrCreateRemoteServiceAsync()
             return default;
         }
 
-        private async Task<TResult> Async<TResult>(Action<Service, RemoteAsyncOperation<TResult>> action)
+        private async Task<TResult> Async<TResult>(string targetName, params object[] arguments)
         {
+            var initializedService = await TryGetOrCreateRemoteServiceAsync().ConfigureAwait(false);
+            TaskCompletionSource<TResult> _completion = new TaskCompletionSource<TResult>();
+            EventHandler _processExitedHandler = new EventHandler((_, __) =>
+            {
+                initializedService.ServiceOpt.Process.Exited -= _processExitedHandler;
+                _completion.TrySetResult(default(TResult));
+            });
+
             try
             {
-                var initializedService = await TryGetOrCreateRemoteServiceAsync().ConfigureAwait(false);
-                if (initializedService.ServiceOpt == null)
+                initializedService.ServiceOpt.Process.Exited += _processExitedHandler;
+                if (!initializedService.ServiceOpt.Process.IsAlive())
                 {
-                    return default;
+                    initializedService.ServiceOpt.Process.Exited -= _processExitedHandler;
+                    _completion.TrySetResult(default(TResult));
                 }
 
-                return await new RemoteAsyncOperation<TResult>(initializedService.ServiceOpt).ExecuteAsync(action).ConfigureAwait(false);
+                return await initializedService.ServiceOpt._jsonRpc.InvokeAsync<TResult>(targetName, arguments).ConfigureAwait(false);
             }
-            catch (Exception e) when (FatalError.Report(e))
+            catch (RemotingException) when (!initializedService.ServiceOpt.Process.IsAlive())
```

>RemotingException [](start = 19, length = 17)

Here we want to catch all exceptions `(catch Exception)`.